homeassistant/components/auth/login_flow.py

```diff
@@ -146,12 +152,63 @@ async def get(self, request: web.Request) -> web.Response:
                 message_code="onboarding_required",
             )
 
-        return self.json(
-            [
-                {"name": provider.name, "id": provider.id, "type": provider.type}
-                for provider in hass.auth.auth_providers
-            ]
-        )
+        try:
+            remote_address = ip_address(request.remote)  # type: ignore[arg-type]
+        except ValueError:
+            return self.json_message(
+                message="Invalid remote IP",
+                status_code=HTTPStatus.BAD_REQUEST,
+                message_code="invalid_remote_ip",
+            )
+
+        cloud_connection = is_cloud_connection(hass)
+
+        providers = []
+        for provider in hass.auth.auth_providers:
+            additonal_data = {}
+
+            if provider.type == "trusted_networks":
+                if cloud_connection:
+                    # Skip quickly as trusted networks are not available on cloud
+                    continue
+
+                try:
+                    cast("TrustedNetworksAuthProvider", provider).async_validate_access(
+                        remote_address
+                    )
+                except InvalidAuthError:
+                    # Not a trusted network, so we don't expose that trusted_network authenticator is setup
+                    continue
+            elif (
+                provider.type == "homeassistant"
+                and not cloud_connection
+                and is_local(remote_address)
+                and "person" in hass.config.components
+            ):
+                # We are local, return user id and username
+                users = await provider.store.async_get_users()
+                additonal_data = {
```

Comment:
No need to create a new dictionary. You can assign a key to the existing `additional_data` dict for this loop iteration.